magda-content-api/src/createApiRouter.ts

```diff
@@ -45,64 +71,63 @@ export default function createApiRouter(options: ApiRouterOptions) {
      *         "result": "FAILED"
      *    }
      */
-    router.get("/:contentId.:format", async function(req, res) {
-        const contentId = req.params.contentId;
-        const format = req.params.format;
+    router.get("/:contentId.:format", getContent);
+    router.get("/*", getContent);
+
+    async function getContent(req: any, res: any) {
+        const requestContentId = req.params.contentId;
+        const requestFormat = req.params.format;
         try {
-            const maybeContent = await database.getContentById(contentId);
-            const content = maybeContent.caseOf({
-                nothing: () => null,
-                just: x => x
-            });
+            const contentPromise = await database.getContentById(
```

Comment:
I've rewritten this part to get rid of the mutability because I wasn't sure how to put it into a comment, let me know if it's ok.

Reply:
Ok. This does look cleaner =)